ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSmain.cpp

```diff
@@ -248,7 +253,7 @@ int WINAPI WinMain (HINSTANCE hInstance,HINSTANCE hPrevInstance,LPSTR lpCmdLine,
     // This will be handled by game_start in roomsystem.cpp where window_default(true) sets the initial
     // fullscreen state of the window before showing it.
     enigma::hWnd = CreateWindow("EnigmaDevGameMainWindow", "",
-                                (enigma::getwindowstyle() & ~(WS_VISIBLE)),
+                                ComputeInitialWindowStyle(),
```

Comment:
Right here is one place you screwed up, we do not want the window immediately visible or we get the brief black window that @time-killer-games reported in #1105 that I fixed in #1113.

So make the first line of `ComputeInitialWindowStyle to the following:
```cpp
LONG_PTR newlong = 0;
```

Reply:
We should definitely have a splash screen, the user needs feedback to know that the program has actually opened, especially if it's a particularly large game, else you end up with people opening 10 different instances of the same game.

Reply:
@faissaloo I agree, I kind of prefer the splash screen over the loading bar. Nobody really does loading bars anymore, they really aren't helpful at all, in addition to looking really ugly. You can glean more information about why a game hypothetically fails using GDB than you do from any log printing out the messages on the load bar, etc.

Also rarely does loading resources ever fail. In fact, I think we've never had a single issue reported about the unpacking mechanism failing (knock on wood). But also consider the fact that you can still technically add a debug message to the splash screen too, in the same way as the loading bar.

The only thing that bothers me about splash screens is they offer the user little control over the sequence. The user can gain more control over the sequence by simply loading all resources externally themselves through code, but that is tedious. What I think we should consider in addition to the splash screens is continuing to fire draw events or adding in additional events to allow users the animate these sequences programmatically in an easy way where the engine is still taking care of all the loading for you.